python/nvtx/nvtx.py

```diff
@@ -37,8 +39,12 @@ def __init__(self, message=None, color=None, domain=None, category=None):
             A message associated with the annotated code range.
             When used as a decorator, the default value of message
             is the name of the function being decorated.
-            When used as a context manager, the default value
-            is the empty string.
+            When used as a context manager, the default value is the empty
+            string.
+            Messaged are cached and are registered as Registered Strings
+            in NVTX.
```

Comment:
Let's avoid using the term "Registered String" as the user may not be aware of that.

```suggestion
            Note that messages are registered with NVTX and cached internally,
            so creating a very large number of annotations can noticeably increase
            memory usage.
```

Ditto below

Reply:
The Nvidia Nsight Systems documentation explicitly mentions registered strings as they are required for filtering. Perhaps we can mention these docs? https://docs.nvidia.com/nsight-systems/UserGuide/index.html#example-interactive-cli-command-sequences

Reply:
OK -- that's fair! Just a small typo here then (`Messaged -> Messages`)

Reply:
Resolved typo